src/app.ts

```diff
@@ -23,45 +24,39 @@ const { NODE_ENV } = envConstants;
 
 export const app: Application = express();
 
+app.use((req: Request, res: Response, next: NextFunction) => {
+  consola.debug(`Request Method: ${req.method}, Request URL: ${req.url} , Request Headers: ${res.links}`);
+  next();
+});
+
 if (NODE_ENV !== "test") {
   // Apply Helmet middleware with default options
-  app.use(helmet());
-
-  // Apply CORS middleware with a whitelist (adjust origins as needed)
-  const allowedOrigins = ["http://localhost:3000"];
-  const allowedPathsWithoutOrigin = ["/api/v1/auth/google/callback", "/api/v1/auth/google"];
+  app.use(helmet()); // Remove '*' as it doesn't work with credentials: true
+  const allowedOrigins = [envConstants.CLIENT_URL, envConstants.USER_SERVICE_BASE_URL].filter(Boolean);
 
   const corsOptions = {
     origin: function (origin: string | undefined, callback: (err: Error | null, allow?: boolean) => void) {
-      if (origin === undefined) {
-        callback(new Error(ErrorTypeEnum.enum.ORIGIN_IS_UNDEFINED));
-      } else if (allowedOrigins.indexOf(origin) !== -1) {
-        callback(null, true);
-      } else {
-        callback(new Error(ErrorTypeEnum.enum.ORIGIN_NOT_ALLOWED));
+      // Allow requests with no origin (mobile apps, Postman, etc.)
+      if (origin === undefined || origin === null) {
+        return callback(null, true);
```

Comment:
Don't forget to remove unused constans. Remove `ORIGIN_IS_UNDEFINED` from `error-types.constant.ts`